### PR TITLE
[gui] Always show "Add rule" item in smartplaylisteditor rule list

### PR DESCRIPTION
**Pull request by phil65**
This changes the smartplaylisteditor rule list to always show the "add rule" item, that way the separate button for that action can get removed.
This allows us to revert  https://github.com/xbmc/xbmc/commit/fc87499d3dfd95790be7e2df52b094d98b939d6b

@Montellese or @xhaggi for review perhaps?
@ronie, mind doing some quick testing?

### CODE REVIEW

**Comment by ronie** [comment]
works ok, not seeing any issues.

**Comment by phil65** [comment]
Cool, thx for testing.
Would it be fine to get this into Krypton? It doesnt need any skin changes, but skinners are free to remove that redundant "add" button with this PR.

File: xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp [hunk 1]
```diff
@@ -188,9 +188,7 @@ bool CGUIDialogSmartPlaylistEditor::OnMessage(CGUIMessage& message)
 
 void CGUIDialogSmartPlaylistEditor::OnPopupMenu(int item)
 {
-  if (item < 0 || item >= m_ruleLabels->Size())
-    return;
-  if (m_playlist.m_ruleCombination.m_rules.size() == 1 && m_playlist.m_ruleCombination.m_rules[0]->m_field == FieldNone)
+  if (item < 0 || item >= static_cast<int>(m_playlist.m_ruleCombination.m_rules.size()))
```
Comment (xhaggi) [inline xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp]: please cast item to `size_t` instead of down case to int.

**Comment by phil65** [comment]
@xhaggi 	done.

File: xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp [hunk 2]
```diff
@@ -352,15 +348,15 @@ void CGUIDialogSmartPlaylistEditor::UpdateButtons()
   CGUIMessage msgReset(GUI_MSG_LABEL_RESET, GetID(), CONTROL_RULE_LIST);
   OnMessage(msgReset);
   m_ruleLabels->Clear();
-  for (unsigned int i = 0; i < m_playlist.m_ruleCombination.m_rules.size(); i++)
+  for (auto rule: m_playlist.m_ruleCombination.m_rules)
```
Comment (xhaggi) [inline xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp]: `const auto rule&`

**Comment by phil65** [comment]
also done. Thx for reviewing.

**Comment by xhaggi** [comment]
np, I can't runtime test it. We nned to trust @ronie ;)

File: xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp (hunk 2 above)
Comment (phil65) [inline xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp]: you mean "const auto& rule" ?

**Comment by phil65** [comment]
Also included in https://github.com/xbmc/xbmc/pull/10594, but i will keep this one open until i know if this (or the other PR) has a chance to get merged for v17

**Comment by MartijnKaijser** [comment]
I think the other might be too intrusive at this stage (although there will be some more betas) as skins need to be updated for the changes?  I'll let this up to the skinners together to decide.

**Comment by phil65** [comment]
Included in https://github.com/xbmc/xbmc/pull/10594

**Pull request closed without merging**